rtt/transports/corba/RemotePorts.cpp

```diff
@@ -194,6 +189,20 @@ RTT::base::ChannelElementBase::shared_ptr RemoteInputPort::buildRemoteChannelOut
     return corba_ceb;
 }
 
+bool RemoteInputPort::disconnect(PortInterface* port)
+{
+    //just check the type and not relay on
+    RemoteOutputPort *oPort = dynamic_cast<RemoteOutputPort* >(port);
+    if(oPort == NULL){
+        Logger::In in("RemoteInputPort::disconnect(PortInterface& port)");
+        log(Error) << "Port: " << this->getName() << " could not be disconnected from: " << oPort->getName()
+                << " because it is not of type: RemoteOutputPort" << endlog();
+        return false;
+    }
+
+    return dataflow->removeConnection(oPort->getName().c_str(), this->getDataFlowInterface(), this->getName().c_str());
```

Comment:
@doudou, yes I got that. But in the patch I implemented both, disconnecto for RemoteOutput and RemoteInputPort. So since the data flow remove connection method uses a pointer to the data flow oject of "the other" port, to me it looks more natural to do that from the "view" of the output port. Anyway wouldn't it be better to have the code only once instead of twice? 
In the RemoteOutputPort than it is also more natural, that the dataflow object on which the function is called is the local one and the pointer given as the remote dataflow object is the one from the RemoteInputPort. So if just calling it in the RemoteInputPort by calling the function of the OutputPort to disconnect from it, and therefore, prevent to have very similar code at two different positions doesn't look too bad for me. Also then on this side it would directly call the right implementation if the given port is a local one.

However, then I guess it moves the question of how to deal with a "local" InputPort given as parameter to the `RemoteOutputPort.disconnect()` function. Is that even a case that can occur? If so, I do not really see how to handle it other than calling again `disconnect(this)` on the port if it is local hoping it can deal with remote ports than, but could call loops here then, which would be bad. But since the FlowInteface needed for the removeConnection function is not provided by an InputPort and to its connectionManager it can only provide a read only (const) version. 
So what is the right way to go for that?